muspinsim/tests/test_hamiltonian.py

```diff
@@ -47,6 +47,88 @@ def test_evolve(self):
 
         self.assertTrue(np.all(np.isclose(evol[:, 0], 0.5 * np.cos(2 * np.pi * t))))
 
+    def test_evolve_density(self):
+
+        ssys = SpinSystem(["e"])
+        ssys.add_linear_term(0, [1, 0, 0])  # Precession around x
+        H = ssys.hamiltonian
+        rho0 = DensityOperator.from_vectors()  # Start along z
+        t = np.linspace(0, 1, 4)
+
+        self.assertTrue(isinstance(H, Hamiltonian))
+
+        # No operators => return a list of density operators
+        evol = H.evolve(rho0, t)
+
+        print(evol[0].matrix)
+        print(evol[1].matrix)
+        print(evol[2].matrix)
+        print(evol[3].matrix)
+
+        self.assertEqual(len(evol), 4)
+        self.assertTrue(
+            np.allclose(
+                evol[0].matrix.toarray(),
+                np.array(
+                    [
+                        [1.00000000e00 + 0.0j, 2.23711432e-17 + 0.0j],
+                        [0.00000000e00 + 0.0j, 0.00000000e00 + 0.0j],
```

Comment:
Yes you essentially need to install muspinsim in someway e.g. `pip3 install -e .` otherwise it wont compile the C++ code into the muspinsim.cpp library.

Reply:
Ah forgot about the `-e` being a thing, that fixed it. I got as far as running `python -v -m muspinsim` revealing the problem was it using:
```
# possible namespace for /home/kez92691/muspinsim/muspinsim/cpp
```
As opposed to:
```
# extension module 'muspinsim.cpp' loaded from '/home/kez92691/muspinsim/muspinsim/cpp.cpython-38-x86_64-linux-gnu.so'
```
I.E. if I ran it somewhere other than root directory of the repo it worked, but if it was somewhere where it could see `muspinsim/muspinsim/cpp` it seemed to prefer that and fail. Still at least I can run unit tests again now (regardless of my directory), thanks a lot.

Reply:
Ah yes forgot about that part as well, I should make a note of that somewhere in the developer docs.